pkg/resources/openstack/network.go

```diff
@@ -123,15 +146,27 @@ func (os *clusterDiscoveryOS) ListNetwork() ([]*resources.Resource, error) {
 		}
 		resourceTrackers = append(resourceTrackers, portTrackers...)
 
-		resourceTracker := &resources.Resource{
-			Name: network.Name,
-			ID:   network.ID,
-			Type: typeNetwork,
-			Deleter: func(cloud fi.Cloud, r *resources.Resource) error {
-				return cloud.(openstack.OpenstackCloud).DeleteNetwork(r.ID)
-			},
+		if !preExistingNet {
+			resourceTracker := &resources.Resource{
+				Name: network.Name,
+				ID:   network.ID,
+				Type: typeNetwork,
+				Deleter: func(cloud fi.Cloud, r *resources.Resource) error {
+					return cloud.(openstack.OpenstackCloud).DeleteNetwork(r.ID)
+				},
+			}
+			resourceTrackers = append(resourceTrackers, resourceTracker)
+		} else {
+			resourceTracker := &resources.Resource{
+				Name: os.clusterName,
+				ID:   network.ID,
+				Type: typeNetworkTag,
+				Deleter: func(cloud fi.Cloud, r *resources.Resource) error {
+					return cloud.(openstack.OpenstackCloud).DeleteTag(openstack.ResourceTypeNetwork, r.ID, r.Name)
```

Comment:
how we can define dependencies to these resources? Now it is possible that network tag is deleted before all the needed resources are deleted. For instance: you execute kops delete cluster --yes, then cancel that in middle of something. Then run that again, you will see that it will not try to delete subnet/router resources anymore because it cannot find network which have tag, because it was deleted. 

If we could have dependency this step should be executed after subnet deletion

Reply:
Can't we just tag all of the resources? Then we don't have to rely on the network being tagged. Also, we don't need to resolve dependencies that way.

Reply:
all resources does not support tags like loadbalancers, and the another thing is that currently we do not have tags in any of the component. Which means that if we modify this logic heavily then we do not have backwards compatibility